app/controllers/parcels_controller.rb

```diff
@@ -0,0 +1,39 @@
+class ParcelsController < ApplicationController
+  rescue_from ActiveRecord::RecordNotFound, with: :rescue_not_found
+
+  def index
+    if current_user
+      @parcels = current_user.parcels.order(created_at: :desc)
+    end
+  end
+
+  def new
+    @parcel = Parcel.new
+  end
+
+  def create
+    @parcel = Parcel.new(parcel_params)
+    if current_user
+      @parcel.sender = current_user
```

Comment:
`@parcel.sender = current_user if current_user`.
Przy czym jedna uwaga - przypisanie relacji w ten sposób automatycznie wywołuje `save` na obu modelach. Będziesz musiał to przebudować - albo przypisać idki, albo przenieść to przypisanie pod `if @parcel.save`.